src/codec/mod.rs

```diff
@@ -11,6 +11,12 @@ mod deflate;
 mod flate;
 #[cfg(feature = "gzip")]
 mod gzip;
+#[cfg(feature = "lzma")]
+mod lzma;
+#[cfg(feature = "xz")]
+mod xz;
+#[cfg(any(feature = "xz", feature = "lzma"))]
```

Comment:
You can use `cfg(feature = "xz2")` here to simplify it, optional crates are also feature names.

Reply:
Huh, I did not know that. Thanks :)